coderd/workspaces.go

```diff
@@ -92,6 +105,22 @@ func (api *API) workspace(rw http.ResponseWriter, r *http.Request) {
 	))
 }
 
+// @Summary List workspaces
+// @ID get-workspaces
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Workspaces
+// @Param owner query string false "Owner username"
+// @Param template query string false "Template name"
+// @Param name query string false "Workspace name"
+// @Param status query string false "Workspace status"
+// @Param deleted query bool false "Deleted workspaces"
+// @Param has_agent query bool false "Has agent"
```

Comment:
These params need better descriptions. I'm pretty sure name, and owner name accept partial matches i.e. `dean` matches `deansheather`

Reply:
I improved the API method description:

<img width="514" alt="Screenshot 2022-12-15 at 18 59 06" src="https://user-images.githubusercontent.com/14044910/207933953-20d5b776-09cc-402b-97f9-4e2584ad0b42.png">

Let me know if you find it better.

> I'm pretty sure name, and owner name accept partial matches

Does it look like the statement is correct for the workspace name, but not for the owner?

```sql
	-- Filter by owner_name
	AND CASE
		WHEN @owner_username :: text != '' THEN
			owner_id = (SELECT id FROM users WHERE lower(username) = lower(@owner_username) AND deleted = false)
		ELSE true
	END
```

Reply:
I was wrong about the owner field in that case, this is good now 👍

Reply:
👍

Reply:
These need to be updated and be consistent anyway. The some are partial, some are not is super annoying.